devito/arch/compiler.py

```diff
@@ -158,7 +158,7 @@ def __init__(self, **kwargs):
         self.ldflags = ['-shared']
 
         self.include_dirs = []
-        self.libraries = ['m']
+        self.libraries = []
```

Comment:
for safety this should be kept here, because it'll affect all other Compiler subclasses (otherwise more refactoring would be needed -- just trying to avoid dumping work on you :) ) 

then ... ( see comment below)